core/Utils.vala

```diff
@@ -699,8 +699,33 @@ namespace Maya.Util {
      * ical Exportation
      */
 
-    public void save_temp_selected_calendars (){
-        //TODO:Create the code !
+    public void save_temp_selected_calendars () {
+        var calmodel = Model.CalendarModel.get_default ();
+        var events = calmodel.get_events ();
+        var builder = new StringBuilder ();
+        builder.append ("BEGIN:VCALENDAR\n");
+        builder.append ("VERSION:2.0\n");
+        foreach (E.CalComponent event in events) {
+            builder.append (event.get_as_string ());
+        }
+        builder.append ("END:VCALENDAR");
+        string output = builder.str;   
+
+        string file_path = GLib.Environment.get_tmp_dir () + "/calendar.ics";
+        try {
+            var file = File.new_for_path (file_path);
+            if (file.query_exists ()) {
+                file.delete ();
+            }
+            var dos = new DataOutputStream (file.create (FileCreateFlags.REPLACE_DESTINATION));
+            uint8[] data = output.data;
+            long written = 0;
+            while (written < data.length) {
+                written += dos.write (data[written:data.length]);
+            }
+        } catch (Error e) {
+            warning ("%s\n", e.message);
+        }
     }
 
```

Comment:
Wouldn't be simpler to create an empty file and the use FileUtils.set_contents (filename, string, -1)?

Reply:
There is very limited documentation of this on valadoc, although there is a fixed bug on Scratch (https://bugs.launchpad.net/scratch/+bug/1021860) that gives some reasons it may be undesirable. I think my main concern with that would be whether it ensures all data is written before returning as the file may be immediately moved by the contractor (export) which could cause issues if the file isn't fully written, which is why I didn't just write the string using the dos. I was unaware of FileUtils so I am not sure of how it works, and whether it ensures the files are written, particularly with the documentation.

Reply:
Agreed, as suggested by that bug report you should use file.replace_contents () rather than FileUtils.set_contents (). If you use the synchronous version then your concerns should be addressed.